vm/devices/storage/disk_nvme/nvme_driver/src/driver.rs

```diff
@@ -868,12 +885,27 @@ impl<T: DeviceBacking> DriverWorkerTask<T> {
                     .find_map(|(i, issuer)| issuer.get().map(|issuer| (i, issuer)))
                     .unwrap();
 
-                tracing::error!(
-                    cpu,
-                    fallback_cpu,
-                    error = err.as_ref() as &dyn std::error::Error,
-                    "failed to create io queue, falling back"
-                );
+                // Log the error as informational only when there is a lack of
+                // hardware resources from the device.
+                match err {
+                    DeviceError::NoMoreIoQueues(_) => {
+                        tracing::info!(
+                            cpu,
+                            fallback_cpu,
+                            error = &err as &dyn std::error::Error,
```

Comment:
I had to prove to myself that this was indeed correct (as opposed to using a `?` or `%` sigil, or playing games with the format string). 

This means our log will contain "failed to create io queue, falling back, error: no more io queues available, reached maximum 11" . This is reasonable.

Under the covers, this is because the tracing::info! macro will treat each field as a `Value` type, and there is special handling for `Value` of `dyn Error` type.